### DISCUSSION TITLE
请问为什么将 Tunnel 处理器的 client id 都替换为了 tunnel id 呢？

### DISCUSSION THREAD

**Discussion by xinghejd** [go-gost/gost#615] [Q&A] [answered]
其中包括limiter、observer，尤其是 limiter，似乎一开始是放在 connect.go 中实现，那个时候是可以取到每次连接的用户 id 的，但是后来似乎关于 tunnel 相关的都放到了 connector 的阶段去实现了，所以都只剩 tunnel id 了，请问这个改动是有什么额外的考量吗？

**Comment by ginuerzh** [comment, marked as answer]
> 其中包括limiter、observer，尤其是 limiter，似乎一开始是放在 connect.go 中实现，那个时候是可以取到每次连接的用户 id 的，但是后来似乎关于 tunnel 相关的都放到了 connector 的阶段去实现了，所以都只剩 tunnel id 了，请问这个改动是有什么额外的考量吗？

Tunnel级别相比用户级别控制粒度更精细，且更实用。

**Comment by xinghejd** [comment]
OK，只要不是技术上遇到问题就好，我这边可能更关注访问用户的粒度，我已经改完了，只是担心有坑，感谢解答。
2024年11月30日 +0800 09:50 ginuerzh ***@***.***>，写道：
